### DISCUSSION TITLE
Saleor payment gateway

### DISCUSSION THREAD

**Discussion by osher-retailin** [saleor/saleor#10057] [Q&A]
I'm trying to use the checkoutComplete store_source to store the new payment source on the payment gateway service.
I'm using Authorize.net and it saves only for the first time when the customer profile is created on the gty.
The next time I want to complete a checkout with a new source the store_source is not working and the payment is not created on the gty.

Does anyone familiar with this problem?

**Comment by timuric** [comment]
@osher-retailin I would advise you to use this flow instead of `checkoutComplete` which relies on plugins https://docs.saleor.io/docs/3.x/developer/checkout#finalizing-checkout-with-saleor-app
